PaperBlossoms/data/README.md

```diff
@@ -0,0 +1,312 @@
+- During runtime, the application queries the application database, fetching the `*_tr` values.
+
+## Contributing data
+
+Below, I describe how to contribute base data. Customer user data is user-specific, and is not meant to be released with the application. Translations have a [separate process](https://github.com/dashnine/PaperBlossoms/wiki/Translation), and can be provided for both the UI and the database.
+
+Contribution guidelines:
+
+- Never modify `paperblossoms.db` directly. We will only merge PRs after validating JSON data and regenerating `paperblossoms.db`.
+- Coordinate with @dashnine about changes to the application database schema or the JSON schemas. Open [an issue](https://github.com/dashnine/PaperBlossoms/issues/new/choose) or [draft PR](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) to make your intention to do so explicit.
+- It is polite to refresh `enum`s in JSON schemas if you add any data which go into `enum`s.
+
+### Contribute a new instance of an existing entity
+
+Follow these steps if you'd like to add a new school, clan, weapon, or any new instance of an entity already captured in an existing JSON file.
+
+0. Set up the mapping between JSON files and their schemas (see [JSON schemas](#json-schemas)).
+1. Add the new object(s) to the relevant JSON file.
+    - Trigger an autocompletion proposal with <kbd>ctrl</kbd>-<kbd>space</kbd>.
+    - Type characters to narrow down string autocompletion proposals.
+    - Navigate between fields in the snippet with <kbd>tab</kbd>.
+    - Nested JSON objects come with nested snippets, trigger the snippet for a new object with <kbd>ctrl</kbd>-<kbd>space</kbd> once your cursor is within `{}`.
+2. Validate the JSON data with `validate_json.py`.
+3. Generate the application database with `json_to_db.py`.
+
+Optional:
+
+4. If you're adding a new instance of an entity in the list of `enum`s, refresh the `enum`s in the affected JSON schemas with `add_enums.py` using the appropriate option.
+
+See [Scripts](#scripts) above for more detail and example usage for each of the scripts mentioned above.
+
+### Contribute a new entity
+
+1. Write a new JSON schema.
+    - Run `add_default_snippets_to_schema.py`
+    - Modify `add_enums.py` to write `enum`s to the new JSON schema as applicable. Run the updated `add_enums.py`.
+2. Set up a mapping beween the new JSON schema and the new JSON file.
+3. Enter data in the new JSON file.
+    - Trigger an autocompletion proposal with <kbd>ctrl</kbd>-<kbd>space</kbd>.
+    - Type characters to narrow down string autocompletion proposals.
+    - Navigate between fields in the snippet with <kbd>tab</kbd>.
+    - Nested JSON objects come with nested snippets, trigger the snippet for a new object with <kbd>ctrl</kbd>-<kbd>space</kbd> once your cursor is within `{}`.
```

Comment:
"Nested JSON objects come with nested snippets, trigger the snippet"  - I'll be honest. My first thought when reading this is... "....what?"